keanu-project/src/main/java/io/improbable/keanu/algorithms/Statistics.java

```diff
@@ -0,0 +1,39 @@
+package io.improbable.keanu.algorithms;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Statistics {
+
+    private Map<Enum, List<Double>> statistics = new HashMap<>();
```

Comment:
Do we need this to be a Map now that we just have an enum provided at init time?  Presumably we could just make this an array and just mandate they use a "sensible" numbering scheme for their enums?

Reply:
Agreed - that would make it faster. I just think the clarity of having these stored in a map is so much clearer for users who are debugging. It enables nice things like letting them see all the available stats really easily with `statistics.keySet()` and I think this is worth the cost. Remember this is only meant for `debugging` purposes